buildSrc/subprojects/integration-testing/src/main/kotlin/org/gradle/gradlebuild/test/fixtures/TestFixturesPlugin.kt

```diff
@@ -60,46 +55,72 @@ open class TestFixturesPlugin : Plugin<Project> {
         configureAsConsumer()
     }
 
+
+    /**
+     * This mimics what the java-library plugin does, but creating a library of test fixtures instead.
+     */
     private
     fun Project.configureAsProducer() {
+        val testFixtures by java.sourceSets.creating
 
-        configurations {
-            create("outputDirs")
+        java.sourceSets.named<SourceSet>("test") {
+            compileClasspath += testFixtures.output
+            runtimeClasspath += testFixtures.output
+        }
 
-            create("testFixturesCompile") { extendsFrom(configurations["compile"]) }
-            create("testFixturesImplementation") { extendsFrom(configurations["implementation"]) }
-            create("testFixturesRuntime") { extendsFrom(configurations["runtime"]) }
+        val testFixturesJar by tasks.creating(Jar::class) {
```

Comment:
Are we going to get rid of the :: syntax?

Reply:
The `::class` syntax is the canonical way to pass types around in Kotlin, so we won't get rid of it entirely 😄 

See https://github.com/gradle/kotlin-dsl/issues/605